### DISCUSSION TITLE
snippet not working

### DISCUSSION THREAD

**Discussion by xxleyi** [johnlindquist/kit#940] [Errors]
Log info looks good, but not working when I type such as `tag,,`, has already forced repair Kit SDK.
```
[2022-11-04 09:50:22.767] [info] Set snippet: newday,,
[2022-11-04 09:50:22.767] [info] Set snippet: tag,,
[2022-11-04 09:50:22.768] [info] Set snippet: //d
```
OS info:
![image](https://user-images.githubusercontent.com/12626454/199868428-0171ac1b-0228-4138-8eb8-66d452cd9bac.png)

**Comment by johnlindquist** [comment]
@xxleyi Snippets require the "Accessibility Permissions". You'll also need to quit and re-open Kit.app after granting permissions. Please let me know if they're still not working after you've tried that

**Reply by xxleyi** [reply]
I had granted permissions, however I try to undo grant and grant again. Now snippets is working well. Thank you.

**Reply by xxleyi** [reply]
And after my mac sleep for about one hour at noon, snippets not worked again. I tried restart Kit, not working.

**Reply by xxleyi** [reply]
And I I try to undo grant and grant again, not working.

Looks Accessibility Permissions part has some hidden problem?

**Reply by xxleyi** [reply]
I just tried again, it just worked. Looks not very steady.

**Reply by johnlindquist** [reply]
1. Is Kit.app installed in your `/Applications` directory?
2. Can you set your log level to "silly", then if it stops working: check `~/.kit/logs/kit.log` to see if it's still logging what you're typing? (It should look like the screenshot below)

I really appreciate the help, thanks!

![CleanShot 2022-11-04 at 06 53 25](https://user-images.githubusercontent.com/36073/199977165-4adab1e6-6d04-4ac5-a763-d4bf4e38bf36.png)
![CleanShot 2022-11-04 at 06 56 14](https://user-images.githubusercontent.com/36073/199977679-15edaeec-8339-4496-a172-6f67554e5212.png)

**Reply by xxleyi** [reply]
I got it. and good thing is it's still working. If not worked again, I will try look silly log.

**Reply by xxleyi** [reply]
Not worked again and I looked silly log, with some input test, I saw such content:
![image](https://user-images.githubusercontent.com/12626454/200244154-39f84c2b-9eef-4f2f-bfd8-a312549f96f7.png)

Looked `Kit` can only know keydown event, but not the character.

**Reply by xxleyi** [reply]
Extra info: I did find some error log, about scheduled script:
```
[2022-11-07 18:43:41.804] [error] 44109: 🟥 exit 1. Prompt process: xxx.ts
```

This error happened when I locked Mac.

**Reply by xxleyi** [reply]
Tried quit and restart, found Kit happened to be "not responding" in activity monitor. and I forced kill and start again, snippet works again.

**Reply by johnlindquist** [reply]
Thanks for the updates. I could try to provide some more real-time help if you prefer Discord:
https://discord.gg/bnVUx4GQ

I'm working on another build to release this week. I'm updating the libraries that handle that key-tracking logic and I'll try to include some more debug info so we can track down what's happening.

**Reply by hakubo** [reply]
I'm experiencing the same issue. 
Looks like keyboard.type does not work also as I can't even trigger snippets manually if they use this

**Reply by hakubo** [reply]
I think I have found why it does not work for me. I have a custom keyboard layout and it looks like scriptkit gets a character from keyboard position and not actually a character that is being input.

**Reply by johnlindquist** [reply]
@hakubo Is that what loglevel silly shows? I thought it worked sometimes then stopped working

**Reply by johnlindquist** [reply]
Also, I almost have a build ready that will have more logging and it will reset the keyboard watcher each time the machine resumes from sleep (since that seemed to by connected to how @xxleyi described it)

**Reply by hakubo** [reply]
nope, it's something I tested.

but my keyboard setup is probably not normal. I have a programmable keyboard set to output qwerty, and set colemak keyboard layout to colmak-dh in my system keyboard settings.

**Reply by hakubo** [reply]
so it looks like scriptkit sees what keyboard sends to system and not what is actually put on screen.

**Reply by johnlindquist** [reply]
@hakubo thanks, that makes sense. I found a lib that might be a little help. Trying to integrate it now.

**Reply by hakubo** [reply]
snippets started to work for me on 1.36, thanks!
but it appears the same issue is happening for shortcuts - any chance for you to look into that also?
